cms/static/js/views/course_info_update.js

```diff
@@ -27,22 +30,58 @@ define(["js/views/baseview", "codemirror", "js/models/course_update",
             // remove and then add all children
             $(updateEle).empty();
             var self = this;
-            this.collection.each(function (update) {
+            this.collection.each(function (update, index) {
                 try {
                     CourseInfoHelper.changeContentToPreview(
                         update, 'content', self.options['base_asset_url']);
                     // push notification is always disabled for existing updates
                     var newEle = self.template({ updateModel : update, push_notification_enabled : false });
                     $(updateEle).append(newEle);
+                    DateUtils.setupDatePicker("date", self, index);
+                    update.isValid();
                 } catch (e) {
                     // ignore
                 }
             });
             this.$el.find(".new-update-form").hide();
-            this.$el.find('.date').datepicker({ 'dateFormat': 'MM d, yy' });
             return this;
         },
 
+        collectionSelector: function(uid) {
+            return "course-update-list li[name=" + uid + "]";
+        },
+
+        setAndValidate: function(attr, value) {
+            if (attr === 'date') {
+                value = $.datepicker.formatDate("MM d, yy", value);
+            }
+            var targetModel = this.collection.get(this.$currentPost.attr('name'));
+            var prevValue = targetModel.get(attr);
+            if (prevValue === value) { return; }
```

Comment:
Nit: a clean code principle is to avoid multiple points of return. I think an exception to this is error checking at the very top of the method. But in this case, I think the code would be cleaner to do this:

````
if (prevValue !=== value) {
    targetModel.set(attr, value);
    this.validateModel(targetModel);
}
````